generator/api/src/main/java/io/fabric8/maven/generator/api/support/BaseGenerator.java

```diff
@@ -121,8 +122,8 @@ protected String getAlias() {
     }
 
     protected boolean shouldAddDefaultImage(List<ImageConfiguration> configs) {
-        String merge =  getConfigWithSystemFallbackAndDefault(Config.alias, "fabric8.generator.alias", "false");
-        return !containsBuildConfiguration(configs) || Boolean.parseBoolean(merge);
+        String merge =  getConfigWithSystemFallbackAndDefault(Config.alias, "fabric8.generator.alias", "");
+        return !containsBuildConfiguration(configs) || Strings.isNotBlank(merge);
     }
```

Comment:
the check for whether to invoke the base generator or spring boot generator was comparing `fabric8.generator.alias` to be true/false; when its the name of the docker image alias name.

We used this configuration a while back to explicity allow spring boot generator and a custom docker image build to coexist together.

i.e. it was a bad boolean condition - `fabric8.generator.alias` is never true.

We could use some other explicit configuration to enable the spring-boot generator if there's a `<configuration><image>` tag for a separate image and there's a spring boot app?